azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+
+func resourceVideoAnalyzerDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.VideoAnalyzersClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VideoAnalyzerID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("issuing AzureRM delete request for Video Analyzer '%s': %+v", id.Name, err)
+	}
+
+	return nil
+}
+
+func expandVideoAnalyzerStorageAccounts(d *pluginsdk.ResourceData) *[]videoanalyzer.StorageAccount {
+	storageAccountRaw := d.Get("storage_account").([]interface{})[0].(map[string]interface{})
+
+	results := []videoanalyzer.StorageAccount{
+		{
+			ID: utils.String(storageAccountRaw["id"].(string)),
+			Identity: &videoanalyzer.ResourceIdentity{
+				UserAssignedIdentity: utils.String(storageAccountRaw["identity_id"].(string)),
+			},
+		},
+	}
+
+	return &results
+}
+
+func flattenVideoAnalyzerStorageAccounts(input *[]videoanalyzer.StorageAccount) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, storageAccount := range *input {
+		output := make(map[string]interface{})
+
+		if storageAccount.ID != nil {
+			output["id"] = *storageAccount.ID
+		}
+
+		if storageAccount.Identity != nil {
+			output["identity_id"] = *storageAccount.Identity.UserAssignedIdentity
+		}
+
+		results = append(results, output)
+	}
+
+	return results
+}
+
+func expandAzureRmVideoAnalyzerIdentity(d *pluginsdk.ResourceData) (*videoanalyzer.Identity, error) {
+	identityRaw := d.Get("identity").([]interface{})
+	if identityRaw[0] == nil {
+		return nil, nil
+	}
+	identity := identityRaw[0].(map[string]interface{})
+	result := &videoanalyzer.Identity{
+		Type: utils.String(identity["type"].(string)),
+	}
+	var identityIdSet []interface{}
+	if identityIds, exists := identity["identity_ids"]; exists {
+		identityIdSet = identityIds.(*pluginsdk.Set).List()
+	}
+
+	// If type contains `UserAssigned`, `identity_ids` must be specified and have at least 1 element
+	if *result.Type == "UserAssigned" {
+		if len(identityIdSet) == 0 {
+			return nil, fmt.Errorf("`identity_ids` must have at least 1 element when `type` includes `UserAssigned`")
+		}
+
+		userAssignedIdentities := make(map[string]*videoanalyzer.UserAssignedManagedIdentity)
+		for _, id := range identityIdSet {
+			userAssignedIdentities[id.(string)] = &videoanalyzer.UserAssignedManagedIdentity{}
+		}
+		result.UserAssignedIdentities = userAssignedIdentities
+	} else if len(identityIdSet) > 0 {
+		// If type does _not_ contain `UserAssigned` (i.e. is set to `SystemAssigned` or defaulted to `None`), `identity_ids` is not allowed
+		return nil, fmt.Errorf("`identity_ids` can only be specified when `type` includes `UserAssigned`; but `type` is currently %q", *result.Type)
+	}
```

Comment:
this shouldn't be possible since both fields within this block are Required, so I think we can remove this?

```suggestion
	}
```

Reply:
You're right. There was some trial and error here, due to the fact the docs don't say it's required, but when you try the API it does. I originally built it as non-required, so there's some redundancy from that 

Commiting your suggestion :)